vendor-extra/JatsContentBundle/tests/EventListener/BuildView/FrontDisplayChannelContentMetaListenerTest.php

```diff
@@ -0,0 +1,219 @@
+            new IdentityTranslator()
+        );
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov">
+    <article-meta>
+        <article-categories>
+            <subj-group subj-group-type="display-channel">
+                <subject>Display Channel</subject>
+            </subj-group>
+        </article-categories>
+    </article-meta>
+</front>
+XML
+        );
+
+        $event = new BuildViewEvent(
+            $element,
+            new View('@LiberoPatterns/content-meta.html.twig', ['items' => ['type' => 'foo']])
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertSame('@LiberoPatterns/content-meta.html.twig', $view->getTemplate());
+        $this->assertSame(['items' => ['type' => 'foo']], $view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_sets_the_text_argument() : void
+    {
+        $translator = new Translator('es');
+        $translator->addLoader('array', new ArrayLoader());
+        $translator->addResource(
+            'array',
+            ['libero.patterns.content_header.meta.type.label' => 'type label in es'],
+            'es',
+            'messages'
+        );
+
+        $listener = new FrontDisplayChannelContentMetaListener($this->createDumpingConverter(), $translator);
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<jats:front xmlns:jats="http://jats.nlm.nih.gov">
+    <jats:article-meta>
+        <jats:article-categories>
+            <jats:subj-group subj-group-type="display-channel">
+                <jats:subject>Display Channel</jats:subject>
+                <jats:subject>Display Channel 2</jats:subject>
+            </jats:subj-group>
+            <jats:subj-group subj-group-type="display-channel">
+                <jats:subject>Display Channel 3</jats:subject>
+            </jats:subj-group>
+        </jats:article-categories>
+    </jats:article-meta>
+</jats:front>
+XML
+        );
+
+        $context = ['lang' => 'es'];
+
+        $event = new BuildViewEvent(
+            $element,
+            new View('@LiberoPatterns/content-meta.html.twig', ['items' => ['foo' => 'bar']], $context)
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertSame('@LiberoPatterns/content-meta.html.twig', $view->getTemplate());
+        $this->assertEquals(
+            [
+                'items' => [
+                    'foo' => 'bar',
+                    'type' => [
+                        'attributes' => [
+                            'aria-label' => 'type label in es',
+                        ],
+                        'content' => [
+                            'node' => '/jats:front/jats:article-meta/jats:article-categories'
```

Comment:
where is this value used later on?

Reply:
The `node`/`template`/`context` combination is the result of the dumping converter https://github.com/libero/browser/blob/2dfdfda5d68035405db39c8dcf1e3368947d4ac2/vendor-extra/JatsContentBundle/tests/EventListener/BuildView/FrontDisplayChannelContentMetaListenerTest.php#L167 https://github.com/libero/browser/blob/2dfdfda5d68035405db39c8dcf1e3368947d4ac2/vendor-extra/ContentPageBundle/tests/ViewConvertingTestCase.php#L19-L22

This shows what's being converted in https://github.com/libero/browser/blob/2dfdfda5d68035405db39c8dcf1e3368947d4ac2/vendor-extra/JatsContentBundle/src/EventListener/BuildView/FrontDisplayChannelContentMetaListener.php#L49-L53

Reply:
I wanted to get to: individual Listeners are responsible for extracting the relevant data through their own Xpath queries.